gallery_dl/extractor/instagram.py

```diff
@@ -0,0 +1,189 @@
+            has_next_page = timeline['page_info']['has_next_page']
+            shortcodes = [n['node']['shortcode'] for n in timeline['edges']]
+
+            for s in shortcodes:
+                url = '{}/p/{}/'.format(self.root, s)
+                yield self._extract_postpage(url)
+
+            if not has_next_page:
+                break
+
+            end_cursor = timeline['page_info']['end_cursor']
+            variables = '{{"id":"{}","first":12,"after":"{}"}}'.format(
+                user_id,
+                end_cursor,
+            )
+            xigis = '{}:{}'.format(rhx_gis, variables)
+            headers = {
+                "X-Requested-With": "XMLHttpRequest",
+                "X-Instagram-GIS": hashlib.md5(xigis.encode()).hexdigest(),
+            }
+            url = '{}/graphql/query/?query_hash={}&variables={}'.format(
+                self.root,
+                '66eb9403e44cc12e5b5ecda48b667d41',
+                variables,
+            )
+            shared_data = json.loads(self.request(url, headers=headers).text)
+
+
+class InstagramPostpageExtractor(InstagramExtractor):
+    """Extractor for PostPage"""
+    subcategory = "postpage"
+    pattern = [r"(?:https?://)?(?:www\.)?instagram\.com/p/([^/]+)/?"]
+    test = [
+        # GraphImage
+        ("https://www.instagram.com/p/BqvsDleB3lV/", {
+            "keyword": {
+                "comments": int,
+                "height": int,
+                "likes": int,
+                "media_id": "1922949326347663701",
+                "shortcode": "BqvsDleB3lV",
+                "typename": "GraphImage",
+                "username": "instagram",
+                "width": int,
+            }
+        }),
+
+        # GraphSidecar
+        ("https://www.instagram.com/p/BoHk1haB5tM/", {
+            "keyword": {
+                "comments": int,
+                "height": int,
+                "likes": int,
+                "media_id": "1875629777499953996",
+                "shortcode": "BoHk1haB5tM",
+                "typename": "GraphSidecar",
+                "username": "instagram",
+                "width": int,
+            }
+        }),
+
+        # GraphVideo
+        ("https://www.instagram.com/p/Bqxp0VSBgJg/", {
+            "url": "8f38c1cf460c9804842f7306c487410f33f82e7e",
+            "keyword": {
+                "comments": int,
+                "height": int,
+                "likes": int,
+                "media_id": "1923502432034620000",
+                "shortcode": "Bqxp0VSBgJg",
+                "typename": "GraphVideo",
+                "username": "instagram",
+                "width": int,
+            }
+        }),
+    ]
+
+    def __init__(self, match):
+        InstagramExtractor.__init__(self, match)
+        self.shortcode = match.group(1)
+
+    def instagrams(self):
+        url = '{}/p/{}/'.format(self.root, self.shortcode)
+        return [self._extract_postpage(url)]
```

Comment:
Not too important, but you can use a tuple instead of a list. (It's a tiny bit more lightweight)
`[...]` -> `(...,)`

Reply:
I have converted it to a tuple.